src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -352,16 +352,19 @@ public static IEnumerable<INamedTypeSymbol> GetContainingTypes(this ITypeSymbol
 
         public static bool IsAttribute(this ITypeSymbol symbol)
         {
-            for (var b = symbol.BaseType; b != null; b = b.BaseType)
-            {
-                if (b.MetadataName == "Attribute" &&
-                    b.ContainingType == null &&
-                    b.ContainingNamespace != null &&
-                    b.ContainingNamespace.Name == "System" &&
-                    b.ContainingNamespace.ContainingNamespace != null &&
-                    b.ContainingNamespace.ContainingNamespace.IsGlobalNamespace)
+            if (symbol != null)
```

Comment:
Do we want to continue down the pattern of allowing null to be passed to extension methods?